parse/lex.go

```diff
@@ -249,23 +251,46 @@ func defaultStateFn(s *state, e emitter) stateFn {
 	return defaultStateFn
 }
 
-// TODO: implement me w/ test cases :-)
+// TODO: handle error case. panic isn't enough
+//
 // NumberStateFn scans an alphanumeric. ex) 123, 4001, 232
 // After reading Number, it returns DefaultStateFn.
+// number = { decimal_digit }
 func numberStateFn(s *state, e emitter) stateFn {
+	if r := s.peek(); !(r == '+' || r == '-' || ('0' <= r && r <= '9')) {
```

Comment:
this is optional, but we can refactor `!(r == '+' || r == '-' || ('0' <= r && r <= '9'))` in a similar way as `isSpace` or `isAlpha~` func

Reply:
i rather change my logic. it looks better to me! due to added testcase